src/core/vector/qgsvectorlayertemporalproperties.cpp

```diff
@@ -554,23 +546,19 @@ QString QgsVectorLayerTemporalProperties::createFilterString( const QgsVectorLay
     {
       if ( !mStartExpression.isEmpty() && !mEndExpression.isEmpty() )
       {
-        return QStringLiteral( "((%1) %2 %3) AND ((%4) %5 %6)" ).arg( mStartExpression,
-               range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ),
```

Comment:
we need to leave the range.includeEnd() ? QStringLiteral( "<=" ) : QStringLiteral( "<" ), logic here -- see above